### ISSUE TITLE
Cursor over fn return point isn't highlighting all occurences

### ISSUE THREAD

**Issue by KubaP** [rust-lang/rust-analyzer#14813]
**rust-analyzer version**: v0.3.1506
(the `rust-analyzer: Show Version` command in vscode seemingly isn't working)

**rustc version**: rustc 1.69.0 (84c898d65 2023-04-16)

This bug is present within vscode.

When I place the cursor on a function return point (I don't know what the correct terminology is), it highlights all of the return points in a function, apart from return points within a let-else statement:
```rust
fn foo() -> bool {
    if let Ok(()) = Ok(()) { return false; }
    let Some(_) = Some(0) else { return false; };
    return true;
}
```

1. Place cursor on `fn` keyword.
2. See that it highlights the `fn` keyword, as well as the `return` keywords on lines 2 and 4, **but not** on line 3.

![image](https://github.com/rust-lang/rust-analyzer/assets/4247403/a04b6943-fcc0-44cb-b0fc-ca198f0c4c41)

Interestingly enough, if you put the cursor on the `return` keyword that isn't being highlighted, the other return points are still highlighted correctly.

**Comment by Veykril** [comment]
The problem is this part here https://github.com/rust-lang/rust-analyzer/blob/b87ee914fa3f74e762fa3dd24bd1c903b9330abe/crates/ide-db/src/syntax_helpers/node_ext.rs#L55
We only check the initializer, not the optional else part

**Comment by bm-w** [comment]
@Veykril, this issue should be fixed now that you merged #14848.
